src/Core/Group/Provider/DefaultGroupsProviderInterface.php

```diff
@@ -35,4 +35,9 @@ interface DefaultGroupsProviderInterface
      * @return DefaultGroups
      */
     public function getGroups();
+
+    /**
+     * @return array of ids
+     */
+    public function getAllGroupIds();
```

Comment:
it is a BC, right ?

Reply:
Yes, adding a new function in an interface is a Backward Compatibility Break as the implementations will need to be updated in order to avoid errors

Reply:
what ? i've just added a new method. Please explain

Reply:
we must therefore be sure to have added this method in all the classes which implemented this interface to avoid a fatal error 
```php

<?php

interface myInterface{
    public function myOldFunction();
    public function myNewFunction(); // method added later
}


class myClass1 implements myInterface {
    public function myOldFunction() {
        
    }
    
    public function myNewFunction() {
        
    }
    
}

class myClass2 implements myInterface {
    public function myOldFunction() {
        
    }
    
    // Missing myNewFunction() 
    // Fatal error : Class myClass2 contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (myInterface::myNewFunction) in
}

```

See http://sandbox.onlinephpfunctions.com/code/cc2919787331e55673f52f994ca1c80991b1435f

Reply:
Si what do you want me to do ?
I guess there is only one class that implements this interface.

Reply:
@PululuK @matks I've checked and it's ok : there is no other classes implementing DefaultGroupsProviderInterface.

Reply:
@jf-viguier In the PrestaShop software yes. But remember that unfortunately we have not defined any official public API. This means that any class, method, interface from PrestaShop ... is supposed to be part of our public API and we have a SemVer obligation to not introduce BC breaks.

Imagine this scenario:

A developer named Jack has built a module ExtraOrderWipeModule
In the module there is a PHP class that implements DefaultGroupsProviderInterface, this is merged for PrestaShop 1.7.9
We modify the DefaultGroupsProviderInterface and add the new function
The module ExtraOrderWipeModule works with PrestaShop 1.7.8
Then merchand upgrades from PrestaShop 1.7.8 to PrestaShop 1.7.9
Now module ExtraOrderWipeModule breaks because it contains a PHP class that uses `implements DefaultGroupsProviderInterface` but the php class misses one function

This is why we cannot modify interfaces 😢 . Until next major release of PrestaShop...

The good way to do it is to create a new interface.
- create new interface
- modify PrestaShop classes that rely on old interface to use the new interface
- deprecate old interface (= we say to people "warning! we are going to delete this interface soon, it works for now but start the modifications of your code now so that you are ready when we remove this")
- in next major release, delete old interface

But until the major release is done, we will have to continue maintaining both old and new interfaces 😢 . This is the big maintainance burden for open source projects.

Because of this burden, we try to avoid doing this (create new interface and deprecate old interface) as much as possible.

Reply:
"The good way to do it is to create a new interface", & The good way to do it is to create a new interface..
I understand this mess. The main issue is having this useless DefaultGroupsProviderInterface.
So, I've moved getAllGroupIds to GroupDataProvider, it's a best place for it.
I've added a simpler sql request to get all ids to avoid useless join in group_lang.